nototools/generate_dingbats_html.py

```diff
@@ -395,13 +395,87 @@ def _load_targets(target_data, fonts, data_dir, codelist_map):
   return tuple(result)
 
 
-def _load_fonts_and_targets(font_data, target_data, data_dir):
+def _create_codeset_from_expr(expr_list, flag_sets, data_dir, codelist_map):
+  """Processes expr_list in order, building a codeset.
+  See _read_flag_data_from_file for information on expr_list.
+  This can modify flag_sets and codelist_map."""
+
+  result = ()
+  for op, exp in expr_list:
+    if exp not in flag_sets:
+      # its a codelist
+      codes = _load_codelist(exp, data_dir, codelist_map).codeset()
+    else:
+      codes_or_spec = flag_sets[exp]
+      if isinstance(codes_or_spec, (set, frozenset)):
+        codes = codes_or_spec
+      else:
+        # replace the spec with the actual codes
+        if codes_or_spec == None:
+          # we only know about '_emoji_'
+          if exp == '_emoji_':
+            codes = (
+                unicode_data.get_emoji() -
+                unicode_data.get_unicode_emoji_variants('proposed_extra'))
+          else:
+            raise Exception('unknown special codeset "%s"' % exp)
+        else:
+          codes = _load_codelist(
+              codes_or_spec, data_dir, codelist_map).codeset()
+        flag_sets[exp] = codes
+    if op == '|':
+      if not result:
+        # it appers that python 'optimizes' by replacing the lhs by rhs if lhs
+        # is an empty set, but this changes the type of lhs to frozenset...
+        result = set(codes)
+      else:
+        result |= codes
+    elif op == '&':
+      result &= codes
+    else:
+      result -= codes
```

Comment:
Shouldn't you need a guard against an unsupported op (op other than {|, &, -} )? Or, you're fine with  treating anything other than {&, |} as '-'?